src/Views/EthernetPage.vala

```diff
@@ -66,6 +72,21 @@ namespace Network.Widgets {
             base.update ();
 
             state = device.state;
+
+            string? old_visible_name = widgets_stack.get_visible_child_name ();
+            string new_visible_name = "plugged";
+            bool ctrl_switch_sens = true;
+
+            if (state == NM.DeviceState.UNAVAILABLE) {
+                new_visible_name = "unplugged";
+                ctrl_switch_sens = false;
+            }
+
+            status_switch.sensitive = ctrl_switch_sens;
+
+            if (old_visible_name == null || old_visible_name != new_visible_name) {
+                widgets_stack.set_visible_child_name (new_visible_name);
+            }
```

Comment:
That seems a little convoluted. Is there a difference between doing that and doing this:
```suggestion
            if (state == NM.DeviceState.UNAVAILABLE) {
                widgets_stack.visible_child_name = "unplugged";
                status_switch.sensitive = false;
            } else {
                widgets_stack.visible_child_name = "plugged";
                status_switch.sensitive = true;
            }
```